### ISSUE TITLE
FatesHydroWTFMod.F90 build errors on Summit with default IBM XL compiler

### ISSUE THREAD

**Issue by dqwu** [E3SM-Project/E3SM#4184]
This issue is not reproducible with GNU compiler. Error messages:
```
"components/elm/src/external_models/fates/biogeophys/FatesHydroWTFMod.F90", 1513-083 (E) Internal or module function th_sat was not set within the function.
FORTRAN   - Phase Ends;   0.000/  0.010
** fateshydrowtfmod   === End of Compilation 1 ===
```

Steps to reproduce:
```
git clone https://github.com/E3SM-Project/E3SM.git
cd E3SM

git submodule update --init

cd cime/scripts
./create_newcase --case FC5AV1C-L_ne4_ne4 --compset FC5AV1C-L --res ne4_ne4

cd FC5AV1C-L_ne4_ne4

./case.setup

./case.build
```

**Comment by dqwu** [comment]
> I found a similar thread in the Trilinos community: [xsdk-project/xsdk-issues#94](https://github.com/xsdk-project/xsdk-issues/issues/94)
> 
> They seem to think that (E) is not a catastrophic error. Is the model completing its build?
case.build would fail. If possible, you can try to reproduce it on Summit.

**Comment by dqwu** [comment]
@rgknox 
This issue is no longer reproducible with latest E3SM master branch. I think it has been fixed by PR #4289.
